packages/@react-spectrum/tabs/src/Tabs.tsx

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {classNames, useStyleProps} from '@react-spectrum/utils';
+import {CollectionChildren, CollectionElement, DOMProps, Node, Orientation, StyleProps} from '@react-types/shared';
+import {FocusRing} from '@react-aria/focus';
+import {ListState, useSingleSelectListState} from '@react-stately/list';
+import {mergeProps} from '@react-aria/utils';
+import {Picker} from '@react-spectrum/picker';
+import React, {Key, useEffect, useRef, useState} from 'react';
+import {SpectrumTabsProps} from '@react-types/tabs';
+import styles from '@adobe/spectrum-css-temp/components/tabs/vars.css';
+import tabsStyles from './tabs.css';
+import {useHover} from '@react-aria/interactions';
+import {useProviderProps} from '@react-spectrum/provider';
+import {useTab, useTabs} from '@react-aria/tabs';
+
+export function Tabs<T extends object>(props: SpectrumTabsProps<T>) {
+  props = useProviderProps(props);
+  let {
+    orientation = 'horizontal' as Orientation,
+    children,
+    onSelectionChange,
+    isDisabled,
+    isQuiet,
+    density,
+    defaultSelectedItem,
+    overflowMode = 'scrolling',
+    ...otherProps
+  } = props;
+  let ref = useRef<HTMLDivElement>();
+
+  let allKeys = React.Children.map(children, (child: CollectionElement<T>) => child.key);
+  let state = useSingleSelectListState<T>({
+    ...props, 
+    onSelectionChange,
+    defaultSelectedKey: defaultSelectedItem || allKeys[0],
+    disabledKeys: isDisabled ? [...allKeys] : []
+  });
+  
+  let {styleProps} = useStyleProps(otherProps);
+  let {tabListProps, tabPanelProps} = useTabs(props, state, ref);
+  
+  let selected = [...state.collection].find(item => state.selectionManager.selectedKeys.has(item.key));
+  let [selectedTab, setSelectedTab] = useState<HTMLElement>();
+  
+  useEffect(() => {
+    let tabs: HTMLElement[] = Array.from(ref.current.querySelectorAll('.' + styles['spectrum-Tabs-item']));
+    if (overflowMode === 'scrolling') {
+      setSelectedTab(tabs.find(tab => tab.dataset.key === state.selectedKey));
+    }
+    if (overflowMode === 'dropdown') {
+      setSelectedTab(tabs[0]);
+    }
+  }, [props.children, state.selectedKey, overflowMode]);
```

Comment:
With regards to the overflow mode you've implemented here is some code that might be of interest: https://github.com/adobe/react-spectrum/blob/main/packages/@react-spectrum/breadcrumbs/src/Breadcrumbs.tsx#L56-L120. Breadcrumbs does some overflow handling that maybe of use. Not a hard requirement for this pull, just putting it here if you are interested

Reply:
I'll remove this from this PR and cover it in another PR! Thank you.